### ISSUE TITLE
Strict parsing

### ISSUE THREAD

**Issue by sollecitom** [uchuhimo/konf#12]
Great work, I was thinking of using it in my current project. One requirement I have, though, is to allow strict parsing, meaning that if a key appears in the configuration, and this key is not part of the Spec, the application should fail to start.

Ideally, there should be a function, at Config level, returning all the keys that are not part of the schema.

**Comment by uchuhimo** [comment]
@sollecitom I can add an API in `Source` to export all the keys in the configuration, and then compare these keys with those in the current layer of Config to detect illegal keys. I would like to introduce a strict mode switch when loading values from source. When the strict mode is on, loading will fail when illegal keys are detected and all the illegel keys will be reported. Since implementing this function for all kinds of sources at once is unrealistic, I would like to support one or two sources at first. Which kind of source are you using now (JSON or YAML or ...)?

**Comment by sollecitom** [comment]
Hey, thanks for the quick reply. The idea is precisely how I would go and do it. No preference in terms of order, because I'd like to support all formats (we currently use HOCON, but I can use Typesafe directly for that).

If you feel like it's too much work, I can submit a PR when I get a bit of free time (not much these days).

**Comment by uchuhimo** [comment]
This feature has been released in v0.12 and the usage is described in https://github.com/uchuhimo/konf#strict-parsing-when-loading.

By default, Konf extracts desired paths from sources and ignores other unknown paths in sources. If you want Konf to throws exception when unknown paths are found, you can enable `FAIL_ON_UNKNOWN_PATH` feature:

```kotlin
config.enable(Feature.FAIL_ON_UNKNOWN_PATH)
    .from.properties.file("server.properties")
    .from.json.resource("server.json")
```

Then `config` will validate paths from both the properties file and the JSON resource. Furthermore, If you want to validate the properties file only, you can use:

```kotlin
config.from.enable(Feature.FAIL_ON_UNKNOWN_PATH).properties.file("/path/to/file")
    .from.json.resource("server.json")
```

I will close this issue, but further discussions and feedbacks are welcome.
